knora/dsplib/utils/xml_upload.py

```diff
@@ -542,7 +529,8 @@ def _purge_stashed_xml_texts(
         a purged version of stashed_xml_text
     """
     # remove resources that couldn't be uploaded. If they don't exist in DSP, it's not worth caring about their xmltexts
-    if id2iri_mapping:
+    if id2iri_mapping is not None:
```

Comment:
If "XY is not None" is not the same as "XY", there seems to be something wrong. Could you maybe solve this problem at `id2iri_mapping` instead of here? I.e. make sure that `if id2iri_mapping` is the same as `if id2iri_mapping is not None` is actually the same?

Reply:
Okay, I found the problem. `id2iri_mapping` should not be optional. It is in fact always available when this method is executed, so it is best to *always* check if the stashed elements are in `id2iri_mapping`. So I don't need the if-clause any more.